### PR TITLE
docs: add instruction for using OFO and GitOps

### PR DESCRIPTION
**Pull request by thisthat**
## This PR

adds some notes on a shortcoming of the compatibility of OFO with Applications deployed via GitOps. 

### Argo example

While deploying an App that makes use of FF via Argo, flagD was continuously failing with the following error log:
```
error   kubernetes/kubernetes_sync.go:115   error with the initial fetch: featureflagconfigurations.core.openfeature.dev "featureflagconfiguration-sample" is forbidden: User "system:serviceaccount:myawesomeapp:default" cannot get resource "featureflagconfigurations" in API group "core.openfeature.dev" in the namespace
```

After some debugging, we discovered that Argo was changing back the RoleBinding deployed with the Application.
We fixed the issue by adding the following info to the Argo App so it doesn't try to reconcile the Role Bindings changed by OFO:
```
ignoreDifferences:
  - group: "rbac.authorization.k8s.io"
    kind: "ClusterRoleBinding"
    jsonPointers:
    - /subjects  
...
syncOptions:
  - RespectIgnoreDifferences=true
```

### CODE REVIEW

**Comment by toddbaert** [comment]
I think this only impacts the `kubernetes` sync, but after looking through the doc, I don't think we do a great job describing the files/kubernetes/flagd-proxy syncs. I think the best place to put this would be in a section talking about the `kubernetes` sync.

How would you feel about adding this to @Kavindu-Dodan 's improved doc here: https://github.com/open-feature/open-feature-operator/pull/496/files#diff-c282aa25f1dabaaba19197006c4aa6bab00e8d2b2100375c42b41b7cf4bc55cbR14 ?

**Comment by thisthat** [comment]
@toddbaert sure thing! I'll wait until that PR is merged and then build on top of that 👍

File: docs/installation.md
Comment (Kavindu-Dodan) [inline docs/installation.md:25]: +1 for the documentation.

It's worth to add an example for argo with referring to its documentation [1]

[1] - https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/#application-level-configuration

File: docs/installation.md
Comment (beeme1mr) [inline docs/installation.md:1]: Perhaps it would be better to recommend the `flagd proxy` approach when using Argo. That eliminates the need to manipulate the `ClusterRoleBinding`.

Comment (thisthat) [reply docs/installation.md:1]: I looked into the code and the operator always changes the SA of the Pod when mutating the resource. I will also do a follow-up to perform that operation only while injecting a K8s source

**Comment by thisthat** [comment]
@toddbaert @Kavindu-Dodan @beeme1mr the PR is ready for review.

File: docs/installation.md
Comment (thisthat) [reply docs/installation.md:1]: https://github.com/open-feature/open-feature-operator/pull/498

File: docs/concepts.md
Comment (toddbaert) [inline docs/concepts.md:37]:
```suggestion
**When deploying an application via GitOps, we recommend using the `flagd-proxy` mode, which doesn't suffer from the shortcomings above.**
```

File: docs/concepts.md
Comment (toddbaert) [inline docs/concepts.md:26]:
```suggestion
The `kubernetes`, `grpc` and `flagd-proxy` sync configuration has the advantage of providing near real-time flag updates (on the order of seconds) to the flagd sidecar. 
```

File: docs/concepts.md
Comment (toddbaert) [inline docs/concepts.md:28]:
```suggestion
The `kubernetes` syncs require the flagd sidecar (and consequently the workload pod) to communicate with the 
```

**Review by toddbaert — APPROVED** [review]
Left optional suggestions.

I think this is pretty much perfect, great addition.

File: docs/concepts.md
Comment (Kavindu-Dodan) [inline docs/concepts.md:32]: Prefer this in a new paragraph to give more prominence

File: docs/concepts.md
Comment (Kavindu-Dodan) [reply docs/concepts.md:37]: Also in a new paragraph/line (see current preview - https://github.com/open-feature/open-feature-operator/blob/1e948ea17de6eebe822b50cc3bed0d05ea607482/docs/concepts.md )

**Review by Kavindu-Dodan — APPROVED** [review]

File: docs/concepts.md
Comment (thisthat) [reply docs/concepts.md:32]:
```suggestion

Furthermore, if you're using GitOps to deliver your application, please ensure that the `ClusterRoleBinding` deployed with the application
```

File: docs/concepts.md
Comment (thisthat) [reply docs/concepts.md:37]:
```suggestion

**When deploying an application via GitOps, we recommend using the `flagd-proxy` mode, which doesn't suffer from the shortcomings above.**
```

**Comment by toddbaert** [comment]
@beeme1mr any concerns with merging this now?

**Comment by thisthat** [comment]
@beeme1mr ping

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**
